### DISCUSSION TITLE
Sudden problem running tests

### DISCUSSION THREAD

**Discussion by projectitis** [flame-engine/flame#2685] [General]
Hi all,

I had been running tests fine previously, e.g.:

```
flutter test .\packages\flame\test\sprite_test.dart
```

However, just recently, when I try this exact same command I get a whole bunch of errors. Basically it can't compile the projects.

```
Error: Couldn't resolve the package 'flame' in 'package:flame/cache.dart'.
Error: Couldn't resolve the package 'flame' in 'package:flame/flame.dart'.
Error: Couldn't resolve the package 'flame_test' in 'package:flame_test/flame_test.dart'.
Error: Couldn't resolve the package 'flutter' in 'package:flutter/services.dart'.
Error: Couldn't resolve the package 'flutter_test' in 'package:flutter_test/flutter_test.dart'.
Error: Couldn't resolve the package 'mocktail' in 'package:mocktail/mocktail.dart'.
```

I have run melos bootstrap.

What step(s) have I missed?

**Comment by projectitis** [comment]
It was bound to be user error (and it was).
You cannot run tests (well, at least I couldn't) from here:

```
flutter/flame/ > flutter test ./packages/flame/test/path_to_test.dart
```

You have to be within the package directory:

```
flutter/flame/packages/flame/ > flutter test ./test/path_to_test.dart
```

**Comment by projectitis** [comment]
Also just a note for future googlers: Golden tests may fail on different systems because of slight variations in the renderer between systems. The golden files need to be generated on linux in order to match the runners in gitlab.

I use Windows, and many of the golden tests fail locally because the golden files are generated on linux.

**Reply by ufrshubham** [reply]
Sometimes even some non golden tests fail on Windows. I've noticed this with one asset loading test.

**Reply by projectitis** [reply]
Yes, I've hit that one too. I followed the code for that one and it's trying to load an asset where it's supposed to be loading from a file. I don't actually know how it passes on other platforms.
